library/general/src/lib/ui/srv_status_component.rb

```diff
@@ -0,0 +1,236 @@
+        "Un-check it to disable the service. "\
+        "This does not affect the current status of the service in the already "\
+        "running system.</p>\n"
+      )
+    end
+
+  protected
+
+    attr_reader :id_prefix
+
+    # Checks if the service is currently running
+    #
+    # Must be redefined for services not following standard procedures
+    #
+    # @return [Boolean]
+    def service_running?
+      Yast::Service.active?(@service_name)
+    end
+
+    # Checks if the service is currently enabled on boot
+    #
+    # Must be redefined for services not following standard procedures
+    #
+    # @return [Boolean]
+    def service_enabled?
+      Yast::Service.enabled?(@service_name)
+    end
+
+    # Starts the service inmediatly
+    #
+    # Must be redefined for services not following standard procedures
+    def start_service
+      log.info "Default implementation of SrvStatusComponent#start_service for #{@service_name}"
+      Yast::Service.Start(@service_name)
+    end
+
+    # Stops the service inmediatly
+    #
+    # Must be redefined for services not following standard procedures
+    def stop_service
+      log.info "Default implementation of SrvStatusComponent#stop_service for #{@service_name}"
+      Yast::Service.Stop(@service_name)
+    end
+
+    # Reloads the configuration of a running service
+    #
+    # Must be redefined for services not following standard procedures
+    def reload_service
+      log.info "Default implementation of SrvStatusComponent#reload_service for #{@service_name}"
+      Yast::Service.Reload(@service_name)
+    end
+
+    # Widget displaying the status and associated buttons
+    def status_widget
+      Left(
+        HBox(
+          Label(_("Current status:")),
+          Label(" "),
+          *label_and_action_widgets
+        )
+      )
+    end
+
+    # Widget to configure the status on boot
+    def on_boot_widget
+      Left(
+        CheckBox(
+          Id("#{id_prefix}_enabled"),
+          Opt(:notify),
+          _("Start During System Boot"),
+          @enabled
+        )
+      )
+    end
+
+    # Widget to configure reloading of the running service
+    def reload_widget
+      opts = [:notify]
+      opts << :disabled unless service_running?
+      Left(
+        CheckBox(
+          Id("#{id_prefix}_reload"),
+          Opt(*opts),
+          _("Reload After Saving Settings"),
+          service_running? && @reload
+        )
+      )
+    end
+
+    def label_and_action_widgets
+      if service_running?
+        [
+          # TRANSLATORS: status of a service
+          Label(_("running")),
+          Label(" "),
+          PushButton(Id("#{id_prefix}_stop"), _("Stop now"))
+        ]
+      else
+        [
+          # TRANSLATORS: status of a service
+          Label(_("stopped")),
+          Label(" "),
+          PushButton(Id("#{id_prefix}_start"), _("Start now"))
+        ]
+      end
+    end
```

Comment:
so basically I see there mixture of two functionalities - widget one ( good ) and service wrappers ( wrong ). I think we should consider using some nicer API for services.